Hive connector will throw No configuration setting found for key 'kerberos_principal"

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

seatunnel2.3.4版本编译后无法连接hive。
集群没有配置kerberos，报错No configuration setting found for key 'kerberos_principal'，
测试当前版本的hive_jdbc没有任何问题，2.3.3及以前版本的hive_source都没有任何问题
集群版本为CDH6.3.2，hive版本是2.1.1

### SeaTunnel Version

2.3.4-release

### SeaTunnel Config

```conf
env {
  execution.parallelism = 1
  job.mode = "BATCH"
}

source {
  Hive {
    table_name="temp.group_brand_order_list_board"
    metastore_uri="thrift://172.xx.xxx.xx:9083"
    hdfs_site_path = "/etc/hadoop/conf/hdfs-site.xml"
  }
}

transform {
}

sink {
    Console {
  }
}
```


### Running Command

```shell
bin/seatunnel.sh --config hive_test.conf -e local
```


### Error Exception

```log
2024-01-29 14:11:56,252 ERROR [o.a.s.c.s.SeaTunnel           ] [main] - Exception StackTrace:org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:199)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.connectors.seatunnel.hive.exception.HiveConnectorException: ErrorCode:[HIVE-02], ErrorDescription:[Initialize hive metastore client failed] - Login form kerberos failed
        at org.apache.seatunnel.connectors.seatunnel.hive.utils.HiveMetaStoreProxy.<init>(HiveMetaStoreProxy.java:101)
        at org.apache.seatunnel.connectors.seatunnel.hive.utils.HiveMetaStoreProxy.getInstance(HiveMetaStoreProxy.java:112)
        at org.apache.seatunnel.connectors.seatunnel.hive.config.HiveConfig.getTableInfo(HiveConfig.java:73)
        at org.apache.seatunnel.connectors.seatunnel.hive.source.HiveSource.prepare(HiveSource.java:124)
        at org.apache.seatunnel.engine.core.parse.JobConfigParser.parseSource(JobConfigParser.java:84)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:313)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:183)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:88)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:161)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:146)
        ... 2 more
Caused by: org.apache.seatunnel.shade.com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'kerberos_principal'
        at org.apache.seatunnel.shade.com.typesafe.config.impl.SimpleConfig.findKeyOrNull(SimpleConfig.java:156)
        at org.apache.seatunnel.shade.com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:174)
        at org.apache.seatunnel.shade.com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:188)
        at org.apache.seatunnel.shade.com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:193)
        at org.apache.seatunnel.shade.com.typesafe.config.impl.SimpleConfig.getString(SimpleConfig.java:250)
        at org.apache.seatunnel.connectors.seatunnel.hive.utils.HiveMetaStoreProxy.<init>(HiveMetaStoreProxy.java:67)
        ... 11 more

2024-01-29 14:11:56,253 ERROR [o.a.s.c.s.SeaTunnel           ] [main] -
===============================================================================



Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:199)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.connectors.seatunnel.hive.exception.HiveConnectorException: ErrorCode:[HIVE-02], ErrorDescription:[Initialize hive metastore client failed] - Login form kerberos failed
        at org.apache.seatunnel.connectors.seatunnel.hive.utils.HiveMetaStoreProxy.<init>(HiveMetaStoreProxy.java:101)
        at org.apache.seatunnel.connectors.seatunnel.hive.utils.HiveMetaStoreProxy.getInstance(HiveMetaStoreProxy.java:112)
        at org.apache.seatunnel.connectors.seatunnel.hive.config.HiveConfig.getTableInfo(HiveConfig.java:73)
        at org.apache.seatunnel.connectors.seatunnel.hive.source.HiveSource.prepare(HiveSource.java:124)
        at org.apache.seatunnel.engine.core.parse.JobConfigParser.parseSource(JobConfigParser.java:84)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:313)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:183)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:88)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:161)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:146)
        ... 2 more
Caused by: org.apache.seatunnel.shade.com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'kerberos_principal'
        at org.apache.seatunnel.shade.com.typesafe.config.impl.SimpleConfig.findKeyOrNull(SimpleConfig.java:156)
        at org.apache.seatunnel.shade.com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:174)
        at org.apache.seatunnel.shade.com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:188)
        at org.apache.seatunnel.shade.com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:193)
        at org.apache.seatunnel.shade.com.typesafe.config.impl.SimpleConfig.getString(SimpleConfig.java:250)
        at org.apache.seatunnel.connectors.seatunnel.hive.utils.HiveMetaStoreProxy.<init>(HiveMetaStoreProxy.java:67)
        ... 11 more
```


### Zeta or Flink or Spark Version

Zeta

### Java or Scala Version

1.8.0_181

### Screenshots

<img width="1095" alt="image" src="https://github.com/apache/seatunnel/assets/155407116/d2a8a67b-c7f0-441e-a17f-80e3646d488d">


### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java
+++ b/seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java
@@ -250,7 +250,7 @@ public class HadoopFileSystemProxy implements Serializable, Closeable {
             return;
         }
         this.fileSystem = FileSystem.get(configuration);
-        fileSystem.setWriteChecksum(false);
+        this.fileSystem.setWriteChecksum(false);
     }
 
     private Configuration createConfiguration() {
@@ -269,40 +269,41 @@ public class HadoopFileSystemProxy implements Serializable, Closeable {
     }
 
     private boolean enableKerberos() {
-        boolean kerberosPrincipalEmpty = StringUtils.isEmpty(hadoopConf.getKerberosPrincipal());
-        boolean kerberosKeytabPathEmpty = StringUtils.isEmpty(hadoopConf.getKerberosKeytabPath());
-        boolean krb5FilePathEmpty = StringUtils.isEmpty(hadoopConf.getKrb5Path());
-        if (kerberosKeytabPathEmpty && kerberosPrincipalEmpty && krb5FilePathEmpty) {
+        boolean kerberosPrincipalEmpty = StringUtils.isBlank(hadoopConf.getKerberosPrincipal());
+        boolean kerberosKeytabPathEmpty = StringUtils.isBlank(hadoopConf.getKerberosKeytabPath());
+        if (kerberosKeytabPathEmpty && kerberosPrincipalEmpty) {
             return false;
         }
-        if (!kerberosPrincipalEmpty && !kerberosKeytabPathEmpty && !krb5FilePathEmpty) {
+        if (!kerberosPrincipalEmpty && !kerberosKeytabPathEmpty) {
             return true;
         }
         if (kerberosPrincipalEmpty) {
             throw new IllegalArgumentException("Please set kerberosPrincipal");
         }
-        if (kerberosKeytabPathEmpty) {
-            throw new IllegalArgumentException("Please set kerberosKeytabPath");
-        }
-        throw new IllegalArgumentException("Please set krb5FilePath");
+        throw new IllegalArgumentException("Please set kerberosKeytabPath");
     }
 
     private void initializeWithKerberosLogin() throws IOException, InterruptedException {
-        HadoopLoginFactory.loginWithKerberos(
-                configuration,
-                hadoopConf.getKrb5Path(),
-                hadoopConf.getKerberosPrincipal(),
-                hadoopConf.getKerberosKeytabPath(),
-                (configuration, userGroupInformation) -> {
-                    this.userGroupInformation = userGroupInformation;
-                    this.fileSystem = FileSystem.get(configuration);
-                    return null;
-                });
+        Pair<UserGroupInformation, FileSystem> pair =
+                HadoopLoginFactory.loginWithKerberos(
+                        configuration,
+                        hadoopConf.getKrb5Path(),
+                        hadoopConf.getKerberosPrincipal(),
+                        hadoopConf.getKerberosKeytabPath(),
+                        (configuration, userGroupInformation) -> {
+                            this.userGroupInformation = userGroupInformation;
+                            this.fileSystem = FileSystem.get(configuration);
+                            return Pair.of(userGroupInformation, fileSystem);
+                        });
         // todo: Use a daemon thread to reloginFromTicketCache
+        this.userGroupInformation = pair.getKey();
+        this.fileSystem = pair.getValue();
+        this.fileSystem.setWriteChecksum(false);
+        log.info("Create FileSystem success with Kerberos: {}.", hadoopConf.getKerberosPrincipal());
     }
 
     private boolean enableRemoteUser() {
-        return StringUtils.isNotEmpty(hadoopConf.getRemoteUser());
+        return StringUtils.isNotBlank(hadoopConf.getRemoteUser());
     }
 
     private void initializeWithRemoteUserLogin() throws Exception {
@@ -314,7 +315,9 @@ public class HadoopFileSystemProxy implements Serializable, Closeable {
                             final FileSystem fileSystem = FileSystem.get(configuration);
                             return Pair.of(userGroupInformation, fileSystem);
                         });
+        log.info("Create FileSystem success with RemoteUser: {}.", hadoopConf.getRemoteUser());
         this.userGroupInformation = pair.getKey();
         this.fileSystem = pair.getValue();
+        this.fileSystem.setWriteChecksum(false);
     }
 }

--- a/seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/utils/HiveMetaStoreProxy.java
+++ b/seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/utils/HiveMetaStoreProxy.java
@@ -56,7 +56,7 @@ public class HiveMetaStoreProxy {
                 String hiveSitePath = config.getString(HiveConfig.HIVE_SITE_PATH.key());
                 hiveConf.addResource(new File(hiveSitePath).toURI().toURL());
             }
-            if (enableKerberos(config)) {
+            if (HiveMetaStoreProxyUtils.enableKerberos(config)) {
                 this.hiveMetaStoreClient =
                         HadoopLoginFactory.loginWithKerberos(
                                 new Configuration(),
@@ -71,7 +71,7 @@ public class HiveMetaStoreProxy {
                                         new HiveMetaStoreClient(hiveConf));
                 return;
             }
-            if (enableRemoteUser(config)) {
+            if (HiveMetaStoreProxyUtils.enableRemoteUser(config)) {
                 this.hiveMetaStoreClient =
                         HadoopLoginFactory.loginWithRemoteUser(
                                 new Configuration(),
@@ -148,26 +148,5 @@ public class HiveMetaStoreProxy {
             hiveMetaStoreClient.close();
             HiveMetaStoreProxy.INSTANCE = null;
         }
-    }
-
-    private boolean enableKerberos(Config config) {
-        boolean kerberosPrincipalEmpty =
-                config.hasPath(BaseSourceConfigOptions.KERBEROS_PRINCIPAL.key());
-        boolean kerberosKeytabPathEmpty =
-                config.hasPath(BaseSourceConfigOptions.KERBEROS_KEYTAB_PATH.key());
-        if (kerberosKeytabPathEmpty && kerberosPrincipalEmpty) {
-            return false;
-        }
-        if (!kerberosPrincipalEmpty && !kerberosKeytabPathEmpty) {
-            return true;
-        }
-        if (kerberosPrincipalEmpty) {
-            throw new IllegalArgumentException("Please set kerberosPrincipal");
-        }
-        throw new IllegalArgumentException("Please set kerberosKeytabPath");
-    }
-
-    private boolean enableRemoteUser(Config config) {
-        return config.hasPath(BaseSourceConfigOptions.REMOTE_USER.key());
     }
 }

--- a/seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/utils/HiveMetaStoreProxyUtils.java
+++ b/seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/utils/HiveMetaStoreProxyUtils.java
@@ -1,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.hive.utils;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import org.apache.seatunnel.connectors.seatunnel.file.config.BaseSourceConfigOptions;
+
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class HiveMetaStoreProxyUtils {
+
+    public boolean enableKerberos(Config config) {
+        boolean kerberosPrincipalEmpty =
+                config.hasPath(BaseSourceConfigOptions.KERBEROS_PRINCIPAL.key());
+        boolean kerberosKeytabPathEmpty =
+                config.hasPath(BaseSourceConfigOptions.KERBEROS_KEYTAB_PATH.key());
+        if (kerberosKeytabPathEmpty && kerberosPrincipalEmpty) {
+            return true;
+        }
+        if (!kerberosPrincipalEmpty && !kerberosKeytabPathEmpty) {
+            return false;
+        }
+        if (kerberosPrincipalEmpty) {
+            throw new IllegalArgumentException("Please set kerberosPrincipal");
+        }
+        throw new IllegalArgumentException("Please set kerberosKeytabPath");
+    }
+
+    public boolean enableRemoteUser(Config config) {
+        return config.hasPath(BaseSourceConfigOptions.REMOTE_USER.key());
+    }
+}

--- a/seatunnel-connectors-v2/connector-hive/src/test/java/org/apache/seatunnel/connectors/seatunnel/hive/utils/HiveMetaStoreProxyUtilsTest.java
+++ b/seatunnel-connectors-v2/connector-hive/src/test/java/org/apache/seatunnel/connectors/seatunnel/hive/utils/HiveMetaStoreProxyUtilsTest.java
@@ -1,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.hive.utils;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+import org.apache.seatunnel.shade.com.typesafe.config.ConfigFactory;
+
+import org.junit.jupiter.api.Test;
+
+import lombok.SneakyThrows;
+
+import java.io.File;
+import java.net.URL;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class HiveMetaStoreProxyUtilsTest {
+
+    @Test
+    void enableKerberos() {
+        Config config = parseConfig("/hive_without_kerberos.conf");
+        assertFalse(HiveMetaStoreProxyUtils.enableKerberos(config));
+        assertFalse(HiveMetaStoreProxyUtils.enableRemoteUser(config));
+
+        config = parseConfig("/hive_with_kerberos.conf");
+        assertTrue(HiveMetaStoreProxyUtils.enableKerberos(config));
+        assertFalse(HiveMetaStoreProxyUtils.enableRemoteUser(config));
+
+        config = parseConfig("/hive_with_remoteuser.conf");
+        assertTrue(HiveMetaStoreProxyUtils.enableRemoteUser(config));
+    }
+
+    @SneakyThrows
+    private Config parseConfig(String configFile) {
+        URL resource = HiveMetaStoreProxyUtilsTest.class.getResource(configFile);
+        String filePath = Paths.get(resource.toURI()).toString();
+        return ConfigFactory.parseFile(new File(filePath));
+    }
+}

--- a/seatunnel-connectors-v2/connector-hive/src/test/resources/hive_with_kerberos.conf
+++ b/seatunnel-connectors-v2/connector-hive/src/test/resources/hive_with_kerberos.conf
@@ -1,0 +1,23 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+{
+    table_name="temp.group_brand_order_list_board"
+    metastore_uri="thrift://localhost:9083"
+    hdfs_site_path = "/etc/hadoop/conf/hdfs-site.xml"
+    kerberos_principal = "hadoop"
+    kerberos_keytab_path = "/home/hadoop/hadoop.keytab"
+}

--- a/seatunnel-connectors-v2/connector-hive/src/test/resources/hive_with_remoteuser.conf
+++ b/seatunnel-connectors-v2/connector-hive/src/test/resources/hive_with_remoteuser.conf
@@ -1,0 +1,22 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+{
+    table_name="temp.group_brand_order_list_board"
+    metastore_uri="thrift://localhost:9083"
+    hdfs_site_path = "/etc/hadoop/conf/hdfs-site.xml"
+    remote_user = "hadoop"
+}

--- a/seatunnel-connectors-v2/connector-hive/src/test/resources/hive_without_kerberos.conf
+++ b/seatunnel-connectors-v2/connector-hive/src/test/resources/hive_without_kerberos.conf
@@ -1,0 +1,21 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+{
+    table_name="temp.group_brand_order_list_board"
+    metastore_uri="thrift://localhost:9083"
+    hdfs_site_path = "/etc/hadoop/conf/hdfs-site.xml"
+}

